src/stravalib/tests/integration/test_client.py

```diff
@@ -512,19 +514,21 @@ def test_create_activity(
     extra_expected_params,
     expected_exception,
 ):
+    # Should we ensure required argument vals are in the default calls?
```

Comment:
Good question. I created these defaults out of laziness for not having to repeatedly mention the `activity_type` in each test case. But now that we're adding `sport_type`, these defaults are not valid anymore.

I suggest the following:

1. Since you already refactored the validation logic into a separate method, let's make a new test for `_validate_activity_type()` that goes over all possible combinations of provided/not-provided and valid/invalid types.
2. The existing tests for `create_activity()` and `update_activity()` could then be simplified by using single default values for `sport_type`.

Reply:
ahhh ok. this i understand. i'll work on this over the weekend. if i do this - i suspect the comment above that i didn't fully follow will also be addressed. please let me know if i'm incorrect in that assumption!

Reply:
ok got it. see my new test below. i'll work on rewriting the other tests in the next few days. I think i understand how things work now but i'd like your input on my new test first to ensure what i think i understand is indeed correct!!